### DISCUSSION TITLE
Events from ol-context-menu-control

### DISCUSSION THREAD

**Discussion by jerryfaust** [MelihAltintas/vue3-openlayers#325] [Q&A] [answered]
Hello.
I recently updated a year-old prototype using vue3-openlayers. I'm glad to see that the project continues to advance. I had to retool a few things in my code, and one thing I cannot get working is handling events from the context menu (beforeopen, open, etc). Here is what I have within the `ol-map` section of my MapView.vue:
```
    <!-- add context menu control -->
    <ol-context-menu-control ref="contextMenu" :items="contextMenuItems" :width="200" 
      @beforeopen="onBeforeOpenMenu" @open="onOpenMenu" @close="onCloseMenu" @add-menu-entry="onAddMenuEntry" />
```
then, down in the `methods` section, I have the following handlers (internal code removed for brevity):
```
    onBeforeOpenMenu(event) {
      console.log(event);
    },
    onOpenMenu(event) {
      console.log(event);
    },
    onCloseMenu(event) {
      console.log(event);
    },
    onAddMenuEntry(event) {
      console.log(event);
    },

```
I have Vue3-OpenLayers Debug turned on, and breakpoints in the handlers. It appears that the events are not being fired. Is there something I need to change?

Thank you for any assistance.
Jerry.

**Comment by d-koppenhagen** [comment, marked as answer]
Not a bug. Just a missing feature. I am on it.

**Reply by d-koppenhagen** [reply]
Released with [version 6.4.0](https://github.com/MelihAltintas/vue3-openlayers/releases/tag/v6.4.0)

**Reply by jerryfaust** [reply]
Wow. Excellent. Thank you.

**Comment by jerryfaust** [comment]
Hello again. If you don't mind, I have a follow up question.

I have included `ref` definitions as follows:
  `<ol-map ref="map" ... />`, and 
  `<ol-context-menu-control ref="contextMenu" ... />`

Then, down in the `methods` section, I am able to call methods on the `map`, such as `this.map.getCoordinateFromPixel(event.pixel)`.
But, when I try calling any methods on the `contextMenu`, such as `this.contextMenu.enable()`, I get the following error:
`Uncaught TypeError TypeError: this.contextMenu.enable is not a function`

This used to work, but perhaps the rules have changed, and having poked around and tried a variety of things, I have not been able to make it work. I confess I am a newbie to both vue and javascript.

[_Just a follow-up note: It is in the_ `methods` _section that I have my context menu event handlers, which is where I need to reference the_ `contextMenu`_. Perhaps there are other options, such as inline functions of some sort? I will try going down this path a little bit._]

**Second follow-up**: I learned something new, that I can use `event.target` to get the control reference (in this case, the `contextMenu`) inside the handler. So at least for now, I will forge ahead using that technique.

Thank you, and Kind Regards,
Jerry.

**Reply by d-koppenhagen** [reply]
Please have a look at:
https://vue3openlayers.netlify.app/componentsguide/mapcontrols/contextmenu/#methods

You have access to all original methods in the `control` child: e.g. `contextMenu.value.control.enable()`.
In case of the map some methods are exposed separately for easier access.

**Reply by jerryfaust** [reply]
Sorry I missed that. Thank you.
